src/Microsoft.ML.Data/DataLoadSave/EstimatorChain.cs

```diff
@@ -93,7 +93,10 @@ public EstimatorChain<TNewTrans> Append<TNewTrans>(IEstimator<TNewTrans> estimat
 
         /// <summary>
         /// Append a 'caching checkpoint' to the estimator chain. This will ensure that the downstream estimators will be trained against
-        /// cached data. It is helpful to have a caching checkpoint before trainers that take multiple data passes.
+        /// cached data. It is helpful to have a caching checkpoint before trainers or feature engineering that take multiple data passes.
+        /// It is also helpful to have after a slow operation, for example after dataset loading from a slow source or after feature
```

Comment:
>slow operation [](start = 47, length = 14)

This is also true only in case there are multiple passes over the data (since caching is done during the first pass).

Reply:
@yaeldekel wouldn't caching help after applying an expensive transformation, even if that transformation takes only one pass? I'm thinking of a scenario where we transform once, and use the transformed data in multiple places.

Reply:
When you call `AppendCacheCheckpoint`, what happens is that the data gets cached right before calling `Fit()` on the next estimator on the chain. So the only thing that matters is how many passes over the data that `Fit()` call does. If it is 0 or 1, then there is no advantage in caching. If it is more than 1, then there is an advantage. 

Now that I think about it, there may be another case: Suppose you have a transformation that is expensive, and in the estimator chain there are two estimators operating on its output, each does one pass over the data. For example, suppose `pipeline` is a pipeline that creates column `C`, and then you have:
```
pipeline.Append(mlContext.Transforms.NormalizeMinMax("MinMax", "C")).Append(mlContext.Transforms.NormalizeMeanVar("MeanVar", "C"));
```
In that case, there **would** be an advantage to add a cache checkpoint before the `MinMax` estimator, since both of these estimators will need to compute `C`. (although, it doesn't seem that this is a very common case).

---
In reply to: [387409264](https://github.com/dotnet/machinelearning/pull/4899#discussion_r387409264) [](ancestors = 387409264)